### PR TITLE
Allow to change MaxConnectionsPerAddress

### PR DESCRIPTION
**Pull request by shargon**
Closes https://github.com/neo-project/neo/issues/630

### CODE REVIEW

**Comment by vncoelho** [comment]
@shargon, it looks good to me, we need to PR on neo-cli as well, right:
```cs
Services/ConsoleServiceBase.cs(293,34): warning CS0168: The variable 'ex' is declared but never used [/opt/neoCli/neo-cli/neo-cli.csproj]
Settings.cs(56,89): error CS0117: 'Peer' does not contain a definition for 'DefaultMinDesiredConnections' [/opt/neoCli/neo-cli/neo-cli.csproj]
Settings.cs(57,75): error CS0117: 'Peer' does not contain a definition for 'DefaultMaxConnections' [/opt/neoCli/neo-cli/neo-c
```

File: neo/protocol.json [hunk 1]
```diff
@@ -1,30 +1,40 @@
...
-      "seed4.neo.org:10333",
-      "seed5.neo.org:10333"
-    ],
-    "SystemFee": {
-      "EnrollmentTransaction": 1000,
-      "IssueTransaction": 500,
-      "PublishTransaction": 500,
-      "RegisterTransaction": 10000
+    "ProtocolConfiguration":
+    {
```
Comment (jsolman) [inline neo/protocol.json]: What was the reason for changing the indentation style for the json?

File: neo/NeoSystem.cs [hunk 2]
```diff
@@ -73,8 +73,7 @@ public void StartConsensus(Wallet wallet, Store consensus_store = null, bool ign
             Consensus.Tell(new ConsensusService.Start { IgnoreRecoveryLogs = ignoreRecoveryLogs }, Blockchain);
         }
 
-        public void StartNode(int port = 0, int wsPort = 0, int minDesiredConnections = Peer.DefaultMinDesiredConnections,
-            int maxConnections = Peer.DefaultMaxConnections)
+        public void StartNode(int port, int wsPort, int minDesiredConnections, int maxConnections)
```
Comment (jsolman) [inline neo/NeoSystem.cs]: You are now passing `maxConnections` as a required param, won’t this always overwrite the default max connections from the confit? Same with `minDesiredConnections`?

**Comment by erikzhang** [comment]
I agree with @jsolman. We should set `MaxConnectionsPerAddress` in config.json.

File: neo/protocol.json (hunk 1 above)
Comment (shargon) [reply neo/protocol.json]: Visual Studio style ...

**Pull request merged**